assets/app/scripts/extensions/javaLink.js

```diff
@@ -40,25 +40,27 @@ angular.module('openshiftConsoleExtensions', ['openshiftConsole'])
       if (!jolokiaPort) {
         return;
       }
-      var pod = $scope.$eval('pod');
+      var pod = $scope.$eval('podTemplate');
       if (!pod) {
         return;
       }
       var podId = pod.metadata.name;
       var namespace = pod.metadata.namespace;
-      $scope.jolokiaUrl = new URI(ProxyPod(namespace, podId, jolokiaPort.containerPort)).segment('jolokia/').toString();
+      $scope.jolokiaUrl = ProxyPod(namespace, podId, jolokiaPort.containerPort).segment('jolokia/').toString();
       $scope.gotoContainerView = function($event, container, jolokiaUrl) {
         $event.preventDefault();
         $event.stopPropagation();
         var returnTo = window.location.href;
         var title = container.name || 'Untitled Container';
+        var token = LocalStorageUserStore.getToken();
         var targetURI = new URI().path(BaseHref)
                                  .segment('java')
                                  .segment('index.html')
                                  .query({
                                    jolokiaUrl: jolokiaUrl,
                                    title: title,
-                                   returnTo: returnTo
+                                   returnTo: returnTo,
+                                   token: token
```

Comment:
So yeah...  here the token needs to get picked up by hawtio-jmx so that it can be set for the Authorization header.  I *could* maybe just have hawtio-jmx look in localStorage but since it's consumed by multiple projects I'd have to look in a couple different localStorage keys.

What it does do though is grab all of these params and stores them, then re-writes window.location.href with just a stored connection name in a pre-bootstrap task.